src/editor/map/map_context.cpp

```diff
@@ -191,13 +191,23 @@ map_context::map_context(const game_config_view& game_config, const std::string&
 	}
 
 	// 0.3 Not a .map or .cfg file
-	if(!filesystem::ends_with(filename, ".map") && !filesystem::ends_with(filename, ".cfg") && !filesystem::ends_with(filename, ".mask")) {
-		std::string message = _("File does not have .map, .cfg, or .mask extension");
+	if(!filesystem::is_map(filename)
+		&& !filesystem::is_mask(filename)
+		&& !filesystem::is_cfg(filename))
+	{
+		std::string message = VGETTEXT("File does not have $map, $cfg, or $mask extension",
```

Comment:
This seems to be going too far - you're hardcoding the number of extensions into the string; we might add another but for compatibility I assume that we'd never remove an old one without a deprecation cycle, so these variables will always expand to the same strings unless the string itself is changed.

So it makes work for the translators without adding any flexibility from the variables.

Reply:
That's right, what about something similar to this:
https://github.com/wesnoth/wesnoth/blob/081866af7b5d52394ad5eaadedc19a364e0b8b35/src/gui/dialogs/file_dialog.cpp#L200 ?

Reply:
`utils::format_disjunct_list` is the tool for this, although to find C++ usage of that, look for its counterpart `format_conjunct_list` instead.

I feel the best option is to just leave the string as-is, and accept that it will need to be changed if more extensions are added.

Reply:
reverted the message to previous version.